### ISSUE TITLE
Re-enable "btrbk tree" command

### ISSUE THREAD

**Issue by GrahamCobb** [digint/btrbk#54]
I hope you don't mind a question rather than a bug here (you might want to consider adding it to the FAQ).

I have been using btrbk for a few months now -- thank you for your efforts.  I have just upgraded my debian stretch system and my cron job fails because the "tree" command has been removed.

Can you suggest any way to get similar information with the new commands?  I can't find a "list" option which shows a similar approach.  "list snapshots" is the closest, but I found the tree formatting to be really useful to see which of my backups was generating the most snapshot activity.

Actually, I don't really need the full "tree" output.  I would really like to see, for each source_subvol, how many snapshot_subvol's there are for it, and the name of the most recent one (whether it is up to date or not would be useful as well).

Any suggestion on getting that info would be great.

**Comment by digint** [comment]
As you already mentioned, "list snapshots" is the closest. It actually shows the same data as the "tree" command in v0.20.0, but in a table format instead of a tree (you can see the up-to-date flag in the status column).

After having implemented all those nice tables, I kind of forgot how much more readable the "tree" output was compared to the more data-analysis-style "list subvolumes". So that's basically why I decided to remove the "tree" command. I used to append the "tree" output in my status emails, but after some months of usage it got so big that it did no make much sense for me any more, but this is of course not necessarily true for others.

Furthermore, the "tree" command did not really print the whole tree (more specifically: it does not show the orphaned backups), that's why it got renamed to "list snapshots", making it clear for the user that he should run "list backups" to get the full story.

> Actually, I don't really need the full "tree" output. I would really like to see, for each source_subvol, how many snapshot_subvol's there are for it, and the name of the most recent one (whether it is up to date or not would be useful as well).

I agree that something like this would make much sense, as many times you are not interested in the backups, but only the snapshots, in order to get a quick overview of your source systems.
The same would also make some sense for the backup side.

That said, I'm wondering how such commands should be named. e.g. `btrbk tree snapshots` and `btrbk tree backups`? Sounds a bit confusing...
Or maybe better rename current `btrbk list snapshots/backups` to `btrbk resolve snapshots/backups`, and have `btrbk list snapshots` just print the snapshot list, without its related backups? Any ideas?

**Comment by GrahamCobb** [comment]
In the meantime, I have created a quick and dirty awk script to get the information I want from the "`list snapshots --format=long`" output.  It doesn't handle spaces in filenames (a real solution would use the "raw" format output) but it meets my immediate need. Included below in case it is useful to anyone else.

```awk
/^snapshot/ {
    if ($3 != current_source_path) {
        if (current_source_path != "") print "Source path", current_source_path, "has", snapshots, "snapshots", status;
        current_source_path = $3;
        status = "";
        snapshots = 0;
    }
    snapshots++; total_snapshots++;
    if ($5 != "-") status = "and is " $5;
}
END {
    if (current_source_path != "") print "Source path", current_source_path, "has", snapshots, "snapshots", status;
    print "Total snapshots =", total_snapshots;
}
```

**Comment by GrahamCobb** [comment]
Thanks for your reply (which I saw after I posted my comment above).

I may not be the best person to comment on naming as I only use the snapshot functionality for now ("list backups" displays nothing for me).  However, my thought was to add a "--format=tree" (for the old tree output) and/or a "--format=summary" (for the sort of summary my awk script gives).  Maybe those could work both for snapshots and for backups.

In any case, thanks for thinking about this!
